modules/pool/lib/staking/fantom/reliquary-staking.service.ts

```diff
@@ -0,0 +1,125 @@
+
+    public async syncStakingForPools(): Promise<void> {
+        const { reliquary } = await this.reliquarySubgraphService.getReliquary({ id: networkConfig.reliquary.address });
+        if (!reliquary) {
+            throw new Error(`Reliquary with id ${networkConfig.reliquary.address} not found in subgraph`);
+        }
+        const farms = await this.reliquarySubgraphService.getAllFarms({});
+        const pools = await prisma.prismaPool.findMany({
+            include: { staking: { include: { farm: { include: { rewarders: true } } } } },
+        });
+        const operations: any[] = [];
+
+        for (const farm of farms) {
+            const pool = pools.find((pool) => isSameAddress(pool.address, farm.poolTokenAddress));
+
+            if (!pool) {
+                console.warn(
+                    `Missing pool for farm with id ${farm.pid} with pool token ${farm.poolTokenAddress}. Skipping...`,
+                );
+                continue;
+            }
+
+            const farmId = `${farm.pid}`;
+            const beetsPerSecond = reliquary.emissionCurve.rewardPerSecond;
+
+            if (!pool.staking) {
+                operations.push(
+                    prisma.prismaPoolStaking.create({
+                        data: {
+                            id: `${reliquary.id}-${farm.id}`,
+                            poolId: pool.id,
+                            type: 'RELIQUARY',
+                            address: networkConfig.reliquary.address,
+                        },
+                    }),
+                );
+            }
+
+            operations.push(
+                prisma.prismaPoolStakingReliquaryFarm.upsert({
+                    where: { id: farmId },
+                    create: {
+                        id: farmId,
+                        stakingId: farmId,
+                        name: farm.name,
+                        beetsPerSecond: beetsPerSecond,
+                    },
+                    update: {
+                        beetsPerSecond: beetsPerSecond,
+                        name: farm.name,
+                    },
+                }),
+            );
+
+            for (let farmLevel of farm.levels) {
+                const { allocationPoints, balance, level, requiredMaturity } = farmLevel;
+
+                operations.push(
+                    prisma.prismaPoolStakingReliquaryFarmLevel.upsert({
+                        where: { id: `${farmId}-${level}` },
+                        create: {
+                            id: `${farmId}-${level}`,
+                            farmId,
+                            allocationPoints,
+                            balance,
+                            level,
+                            requiredMaturity,
+                            // apr will be updated by apr service
+                            apr: 0,
+                        },
+                        update: {
+                            allocationPoints,
```

Comment:
allocation points never change for a level, do they?

Reply:
true, thought they can but nope